src/components/Road.vue

```diff
@@ -72,18 +71,14 @@ export default {
       const today = new Date();
       const currentYear = today.getFullYear();
       const baseYear = (today.getMonth() >= 5 && today.getMonth() <= 10) ? currentYear + 1 : currentYear;
-      return baseYear - Math.floor((this.currentSemester - 1) / 3);
+      return baseYear - Math.floor((this.$store.state.currentSemester - 1) / 3);
```

Comment:
maybe `year` would be best as a computed property in the store now, since this `-1)/3` logic is in multiple places?